cloudservice/aws/aws.go

```diff
@@ -49,25 +49,28 @@ func (s *CloudProviderIPRangeManager) ConvertToGenericIPRanges(unparsedIPRanges
 		return nil, err
 	}
 
-	genericIPRanges := types.CloudProviderIPRangeInfo{
-		CloudProviderServiceName: types.CloudProviderAWS,
-		Version:                  ipRanges.SyncToken,
-		IPRanges:                 make(map[string][]string),
+	result := &types.CloudProviderIPRangeInfo{
+		ServiceList: []string{},
+		IPRanges:    make(map[string][]string),
+		Version:     ipRanges.SyncToken,
 	}
 
 	for _, ipRange := range ipRanges.Prefixes {
-		// this service is the list of all ip ranges of all services + misc ones. We don't need that
+		// the service AMAZON is the list of all ip ranges of all services + misc ones. We don't need that
 		// it's also too big for us to be able to filter all ips
-		if ipRange.Service == "AMAZON" {
+		// remove empty IPPrefix for safety
+		if ipRange.Service == "AMAZON" || ipRange.IPPrefix == "" {
 			continue
 		}
 
-		if len(genericIPRanges.IPRanges[ipRange.Service]) == 0 {
-			genericIPRanges.IPRanges[ipRange.Service] = []string{}
+		if len(result.IPRanges[ipRange.Service]) == 0 {
+			result.ServiceList = append(result.ServiceList, ipRange.Service)
+
+			result.IPRanges[ipRange.Service] = []string{}
```

Comment:
This may not be needed as append will create an array if `nil`
Here the goal is to add the service to the service list once only correct?

Reply:
Thanks! I did not know an append would create the array itself